staging/sockets/luasockets.c

```diff
@@ -0,0 +1,205 @@
+/* Lua sockets module */
+
+#include <sys/lua.h>
+#include <sys/module.h>
+#include <sys/socketvar.h>
+#include <netinet/in.h>
+
+#include <lua.h>
+#include <lauxlib.h>
+
+#include <stdio.h>
+
+MODULE(MODULE_CLASS_MISC, luasockets, "lua");
+
+#define BUF_SIZE 1024
+
+struct luasocket {
+	struct socket *socket;
+	int fd;
+};
+
+static int
+sockets_socket(lua_State *L)
+{
+	struct luasocket *luasocket;
+	struct socket *so;
+	const char *domain;
+	const char *type;
+	int sfd;
+	int err;
+	int dom = 0;
+	int typ = 0;
+
+	domain = luaL_checkstring(L, 1);
+	type   = luaL_checkstring(L, 2);
+
+	//XXX not about using strings to name domains (types, etc)
```

Comment:
take a look at: http://w3.impa.br/~diego/software/luasocket/reference.html. Perhaps it's a better idea to mimic this API.